### ISSUE TITLE
ProgressBar keeps running after exception

### ISSUE THREAD

**Issue by philippjfr** [dask/dask#1791]
Using dask 0.12.0 the progress bar keeps running after an exception occurs. Using [this census_sample.h5](http://assets.holoviews.org/census_sample.h5) file, the issue occurs when running:

```python
pbar = ProgressBar()
pbar.register()
df = dd.read_hdf('census_sample.h5', key='census')
dd.io.to_castra(df, './census.castra')
```

**Comment by jbednar** [comment]
Here's a self-contained example needing no sample file:

```
import dask.dataframe as dd
import numpy as np

from dask.diagnostics import ProgressBar
ProgressBar().register()

!touch dummy.castra
x = np.zeros(100)
df=dd.from_array(x)
dd.io.to_castra(df, 'dummy.castra')
```

Just put this in a cell in a Jupyter notebook and execute that cell.  (Run All doesn't do it, only cell by cell.)

**Comment by jcrist** [comment]
`to_castra` uses the synchronous scheduler internally, which would fail to properly close out the callback. Our test used the threaded scheduler, which is how this snuck through. Fixed by #1826.

**Comment by jbednar** [comment]
Thanks!
